### PR TITLE
signup page

### LINKED ISSUE #12: Signup page
### Sign up page 
**path** : /signup
_________
**sign up page consist of :** 

- header
- input fields and labels allow the user to input data
- button to submit data
- footer 

________
image

### PR DESCRIPTION
**Pull request by khamis-hesham**
fix #12 

- [x] create signup form
- [x] add yup library and apply its validation
- [x] Add validation check at finish button so the helper messages will show up based on user input data

### CODE REVIEW

File: client/src/pages/Signup/index.js [hunk 1]
```diff
@@ -1,5 +1,236 @@
...
+  confirmPasswordSchema,
+} from '../../utils';
 
-const Signup = () => <div>Hi from Signup</div>;
+const Signup = () => {
+  const classes = useStyles();
+  const history = useHistory();
+
+  const [firstName, setFirstName] = useState();
+  const [lastName, setLastName] = useState();
```
Comment (mFlaifel) [inline client/src/pages/Signup/index.js:28]: don't use undefined value , use empty string as a default value

File: client/src/pages/Signup/index.js [hunk 2]
```diff
@@ -1,5 +1,236 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
+import CssBaseline from '@material-ui/core/CssBaseline';
```
Comment (mFlaifel) [inline client/src/pages/Signup/index.js:3]: no need for this we use it in the app.js

**Review by imansedky5 — DISMISSED** [review]

File: client/src/utils/signUpValidation.js [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  firstName: yup.string().min(3).required(),
+});
+export const lastNameSchema = yup.object({
+  lastName: yup.string().min(3).required(),
+});
+export const emailSchema = yup.object({
+  email: yup.string().email().required(),
+});
+export const passwordSchema = yup.object({
+  password: yup.string().min(6).required(),
```
Comment (yasmeenattallah) [inline client/src/utils/signUpValidation.js:13]: password min char must be  same as sign-in validation ==> min 8 char

File: client/src/pages/Signup/index.js [hunk 4]
```diff
@@ -1,5 +1,234 @@
-import React from 'react';
+import React, { useState } from 'react';
+import Button from '@material-ui/core/Button';
```
Comment (mFlaifel) [inline client/src/pages/Signup/index.js:2]: suggestion:
we have a custom button in our component, please use it

File: client/src/pages/Signup/index.js [hunk 5]
```diff
@@ -1,5 +1,234 @@
...
+
+  const [firstNameError, setFirstNameError] = useState(false);
+  const [lastNameError, setLastNameError] = useState(false);
+  const [emailError, setEmailError] = useState(false);
+  const [passwordError, setPasswordError] = useState(false);
+  const [confirmPasswordError, setConfirmPasswordError] = useState(false);
+  const [passwordsMismatchError, setPasswordsMismatchError] = useState(false);
+
+  const [loading, setLoading] = useState(false);
+  const [Error, setError] = useState('');
```
Comment (mFlaifel) [inline client/src/pages/Signup/index.js:40]: we use lowercase in useState
```suggestion
  const [error, setError] = useState('');
```

File: client/src/pages/Signup/index.js [hunk 6]
```diff
@@ -1,5 +1,234 @@
...
+
+  const clear = () => {
+    setFirstName('');
+    setLastName('');
+    setEmail('');
+    setPassword('');
+    setConfirmPassword('');
+    setError(null);
```
Comment (mFlaifel) [inline client/src/pages/Signup/index.js:48]: suggestion :
I think default value is better to be the same type. 
```suggestion
    setError('');
```

File: client/src/pages/Signup/index.js [hunk 7]
```diff
@@ -1,5 +1,234 @@
...
+        password,
+      });
+      clear();
+      setLoading(false);
+      return history.push('/');
+    } catch (err) {
+      setError(err.response ? err.response.data.message : err.errors[0]);
+      return setLoading(false);
+    }
+  };
```
Comment (mFlaifel) [inline client/src/pages/Signup/index.js:85]: when I have an error and click on submit button the loading indicator is shown,
please hide it.
image

**Pull request merged**